#157906024: sorting and filtering product search results

chuzksy-codeactive opened the pull request:
#### What does the PR do?
It sorts and filters product search results
#### Description of Task to be completed?
- made the search text-input more interactive
- filter products by price range
- filter products by vendor
- sort products by highest price
- sort products by lowest price
- sort products by latest arrivals
- sort products by highest and lowest ratings
- write end-to-end test for the renderSortFilter() component
#### How should this be manually tested?
- clone the repository and cd into the `thorin-rc` directory
- run `npm install` and then run `reaction` on the terminal
- to run the end-to-end test simply run `nightwatch`
#### Screenshots
<img width="1440" alt="screen shot 2018-06-25 at 4 50 52 pm" src="https://user-images.githubusercontent.com/32555808/41861425-267a2318-7899-11e8-9fcd-e1afe04a629b.png">
<img width="1440" alt="screen shot 2018-06-27 at 4 58 39 am" src="https://user-images.githubusercontent.com/32555808/41952241-df4773ae-79c6-11e8-8300-b739e5cc4af6.png">
<img width="1023" alt="screen shot 2018-06-25 at 5 00 30 pm" src="https://user-images.githubusercontent.com/32555808/41861493-56606c9a-7899-11e8-9c7e-09f9a06857ff.png">
#### What are the relevant pivotal tracker stories?
[Finishes #157906024]

Orlayhemmy committed:
cancel order option

Commit:
feat(digitalProducts): implement purchase of digital products
- ensure user gets download mail after purchase
- ensure admin can edit digital product
- ensure user can download digital product on Reaction commerce
[Finishes #158241793]

Orlayhemmy committed:
merge staging

Orlayhemmy committed:
purchase of digtal products

Commit:
feature: sorting and filtering
- made the search text-input more interactive
- filter products by price range
- filter products by vendor
- sort products by highest price
- sort products by lowest price
- sort products by latest arrivals
- sort products by highest and lowest ratings
- write end-to-end test for renderSortFilter() component
[Finishes #157906024]

Commit:
feature: sorting and filtering
- made the search text-input more interactive
- filter products by price range
- filter products by vendor
- sort products by highest price
- sort products by lowest price
- sort products by latest arrivals
- sort products by highest and lowest ratings
- write end-to-end test for renderSortFilter() component
[Finishes #157906024]

Orlayhemmy committed:
cancelorder

emmabaye reviewed (commented):
I have tested the feature and it is functional. Good job. However, I noticed that setting the filter before searching has no effect unless the search results have been returned. Also, the landing page seems fixed and not scrollable after I exit search feature.

Orlayhemmy committed:
cancel order methods

Orlayhemmy committed:
fix code line length

andela-moe commented:
@chuzksy-codeactive Please review comments and resolve.

Orlayhemmy committed:
Resolve git conflict

Orlayhemmy committed:
fix houndCI error

Orlayhemmy committed:
fix HoundCI error

Commit:
feature: sorting and filtering
- made the search text-input more interactive
- filter products by price range
- filter products by vendor
- sort products by highest price
- sort products by lowest price
- sort products by latest arrivals
- sort products by highest and lowest ratings
- write end-to-end test for renderSortFilter() component
[Finishes #157906024]

Commit:
feature: sorting and filtering
- made the search text-input more interactive
- filter products by price range
- filter products by vendor
- sort products by highest price
- sort products by lowest price
- sort products by latest arrivals
- sort products by highest and lowest ratings
- write end-to-end test for renderSortFilter() component
[Finishes #157906024]

kenoseni committed:
create a branch for demo

kenoseni committed:
Merge pull request #17 from andela/feat/cancel-order/#157906018

Feat/cancel order/#157906018

Commit:
feature: sorting and filtering
- made the search text-input more interactive
- filter products by price range
- filter products by vendor
- sort products by highest price
- sort products by lowest price
- sort products by latest arrivals
- sort products by highest and lowest ratings
- write end-to-end test for renderSortFilter() component
[Finishes #157906024]
